### PR TITLE
[CI] Implement CI workflow to run e2e tests with BATS

### LINKED ISSUE #133: [CI] Incorporate the running of the BATS integration tests in the CI workflow
**Current Behavior**

Currently, BATS tests can only be run locally, they're not executed as part of the CI-workflow.

**Desired Behavior**

BATS integration tests should be executed as part of  GitHub CI-workflow.

---
**Resources**

* Creating cluster using KinD: https://github.com/layer5io/meshery-linkerd/blob/master/.github/workflows/ci.yml
* There exists a github action that could be used: https://github.com/marketplace/actions/setup-bats-testing-framework
* BATS version used in consul is https://github.com/bats-core/bats-core (https://github.com/sstephenson/bats isn't maintained anymore)

**Alternatives / Additional Context**

* The tests are run directly against the adapter.
* Locally, the tests can be run by `make e2e-tests-localhost` if the adapter is reachable on localhost
* If not, the the env variable `MESHERY_ADAPTER_ADDR` can be set, and `make e2e-tests` used to run the tests.

### PR DESCRIPTION
**Pull request by ramrodo**
Signed-off-by: Rodolfo Martínez Vega <rodomar@outlook.com>

**Description**

This PR creates a Github Action workflow to run e2e tests with BATS

This PR fixes #133

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [X] Yes, I signed my commits.

### CODE REVIEW

**Comment by ramrodo** [comment]
@mgfeller could you review this PR, please? I added a workflow to execute the e2e tests using BATS. Let me know what do you think.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -41,4 +41,65 @@ jobs:
...
+  e2e-tests:
+    needs: [golangci, build]
+    name: End-to-end tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Check out meshery code
```
Comment (mgfeller) [inline .github/workflows/ci.yml:59]: Is there any specific reason to use `meshery` to access `meshery-consul`?
locally, I just run `meshery-consul` and run tests directly against it (localhost:10002).

**Review by mgfeller — COMMENTED** [review]
@ramrodo lovely! finally, depandabot PRs _will_ be tested. there were at least 2 patch updates from our libraries recently that broke the  tests and uncovered a problem.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ramrodo) [reply .github/workflows/ci.yml:59]: Oh, there is no reason to have `meshery` so. I was thinking it was necessary to have it.
Let me remove that part.

**Review by mgfeller — APPROVED** [review]
Geat! :smile:

**Comment by ramrodo** [comment]
I am not able to merge this PR. Do anyone of you want to do it?

**Comment by mgfeller** [comment]
> I am not able to merge this PR. Do anyone of you want to do it?

I can do it :-)

**Comment by leecalcote** [comment]
💥

**Pull request merged**